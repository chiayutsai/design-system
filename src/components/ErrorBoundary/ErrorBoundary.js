import withStyles from 'isomorphic-style-loader/withStyles'
import PropTypes from 'prop-types'
import { Component } from 'react'
import { Translation } from 'react-i18next'
import { SYSTEM_ERROR } from 'constants/error'
import styles from './ErrorBoundary.scss'

class ErrorBoundary extends Component {
  constructor(props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch(error, info) {
    // Display fallback UI
    this.setState({ hasError: true })
    // TODO: Log error to server
    console.error(error, info)
  }

  render() {
    const { hasError } = this.state
    const { children } = this.props

    return (
      <Translation>
        {t => (
          <>
            {hasError ? (
              <div className={styles.error}>
                <div>
                  {t('lbl_SysErrorMsg')}[{SYSTEM_ERROR}]
                </div>
              </div>
            ) : (
              children
            )}
          </>
        )}
      </Translation>
    )
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.node.isRequired,
}

export default withStyles(styles)(ErrorBoundary)
