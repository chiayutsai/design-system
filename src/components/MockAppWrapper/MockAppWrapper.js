import history from 'history'
import PropTypes from 'prop-types'
import App from 'components/App'
import configureStore from 'store/configureStore'
import { injectReducer } from 'store/makeRootReducer'

const initialState = {}

export const store = configureStore(initialState, { history })

const MockAppWrapper = ({ children }) => (
  <App
    context={{
      pathname: '',
      store,
      injectReducer,
    }}
    insertCss={() => {}}>
    {children}
  </App>
)

MockAppWrapper.propTypes = {
  children: PropTypes.node.isRequired,
}

export default MockAppWrapper
