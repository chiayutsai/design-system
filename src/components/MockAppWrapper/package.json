{
  "name": "MockAppWrapper",
  "version": "0.0.0",
  "private": true,
  "main": "./MockAppWrapper.js"
}
