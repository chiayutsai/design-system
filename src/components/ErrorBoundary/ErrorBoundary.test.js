import renderer from 'react-test-renderer'
import MockAppWrapper, { store } from 'components/MockAppWrapper'
import setupLocale from 'test/utils/setupLocale'
import ErrorBoundary from './ErrorBoundary'

describe('ErrorBoundary', () => {
  // setup i18next if component needed
  setupLocale(store)

  test('render correctly', () => {
    const wrapper = renderer
      .create(
        <MockAppWrapper>
          <ErrorBoundary>
            <div className="child" />
          </ErrorBoundary>
        </MockAppWrapper>,
      )
      .toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
