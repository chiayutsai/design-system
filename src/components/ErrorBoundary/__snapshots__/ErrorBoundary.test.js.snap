// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary render correctly 1`] = `
<div
  className="child"
/>
`;
